### PR TITLE
DOC: Final reorganization of documentation pages

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] xref #24499

You can see the changes here: https://datapythonista.github.io/pandas-doc-preview/

Pages moved:
- comparison_with_whatever -> getting_started/comparison/comparison_with_whatever
- ecosystem -> getting_started/ecosystem
- cookbook -> user_guide/cookbook
- rpy2 has been removed

Changes to the documentation home page:
- Moved the long description to the Package overview page, and left a small paragraph, and a link to it
- Made the toctree show just 2 levels, so it's useful and it can be seen what it's there

May be useful to see how the master docs look like, to compare: https://pandas-docs.github.io/pandas-docs-travis/

I will follow up with:
- A change of the style (the one we saw in the Austin sprint)
- An integration with the pandas website (pydata.pandas.org). I'll create a issue for this before, as there is some discussion needed.

@TomAugspurger @jorisvandenbossche @jreback

### CODE REVIEW

**Comment by TomAugspurger** [comment]
> rpy2 has been removed

Thoughts on making that a subsection of the "Comparison with R" page? It's not really a comparison, but I imagine that people coming from pandas to R would be interested in it.

File: doc/source/getting_started/overview.rst [hunk 1]
```diff
@@ -6,25 +6,80 @@
...
-  implementation of customized frequencies.
-* Input/Output tools: loading tabular data from flat files (CSV, delimited,
-  Excel 2003), and saving and loading pandas objects from the fast and
-  efficient PyTables/HDF5 format.
-* Memory-efficient "sparse" versions of the standard data structures for storing
-  data that is mostly missing or mostly constant (some fixed value).
-* Moving window statistics (rolling mean, rolling standard deviation, etc.).
+**pandas** is a `Python <https://www.python.org>`__ package providing fast,
+flexible, and expressive data structures designed to make working with
+"relational" or "labeled" data both easy and intuitive. It aims to be the
```
Comment (jorisvandenbossche) [inline doc/source/getting_started/overview.rst]: this is just a copy paste from what was there before, correct? 

(before I start nitpicking on it :-))

Comment (datapythonista) [reply doc/source/getting_started/overview.rst]: Yes, sorry, moved from the home to the overview without any change.

I didn't move the note that being familiar with numpy was required, I didn't feel it made sense in any of the pages.

**Comment by jorisvandenbossche** [comment]
I am not fully sure I find the ecosystem page fitting for the getting started section (it is also relevant for people familiar with pandas that would not look anymore at a getting started section), but +1 on the other changes!

**Comment by datapythonista** [comment]
I didn't feel like the cookbook and the ecosystem belong 100% to any of the sections, but I don't think they are important enough to be in the top level.

I'm also happy moving the ecosystem to the user guide, but won't be able to take care myself of it until late this evening (UK time), I need to go now.

**Comment by jorisvandenbossche** [comment]
> I didn't feel like the cookbook and the ecosystem belong 100% to any of the sections, but I don't think they are important enough to be in the top level.

Yep, agreed. I would maybe rather put it in the user guide then, but no strong feeling.

**Comment by jorisvandenbossche** [comment]
Actually, I think that a "more curated" ecosystem page would be valuable enough for top-level (with content like: want to scale your pandas -> see those packages, extend pandas data types -> look for those, ... and then only random list of packages). But that's for another issue :-)

**Comment by TomAugspurger** [comment]
See https://gitter.im/pydata/pandas?at=5c4afb5dba355012a496295e

@jorisvandenbossche and I both like the idea of moving ecosystem to the top level and then putting some care into it to make it actually nice. We'll followup with the improvements, but move it to the top level for now to avoid the multiple redirects.

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
The doc and lint jobs finished. Merging.

**Comment by jorisvandenbossche** [comment]
Cool!

**Comment by datapythonista** [comment]
Having the ecosystem in the top level, and improving it sounds good to me too.

If there is time, in couple of hours I could try to make the home page links a bit more compact:
```
PDF Version

Zipped HTML

Date: Jan 25, 2019 Version: 0.24.0rc1+73.gbb86a9d3

Binary Installers: https://pypi.org/project/pandas

Source Repository: https://github.com/pandas-dev/pandas

Issues & Ideas: https://github.com/pandas-dev/pandas/issues

Q&A Support: https://stackoverflow.com/questions/tagged/pandas

Developer Mailing List: https://groups.google.com/forum/#!forum/pydata
```
I think something like the next would be not only more compact, but also more clear:
```
Date: Jan 25, 2019 Version: 0.24.0rc1+73.gbb86a9d3

Download documentation: PDF Version | Zipped HTML

Useful links: Binary Installers | Source Repository | Issues & Ideas | Q&A Support | Mailing List
```
 What do you think?

**Comment by TomAugspurger** [comment]
> If there is time, in couple of hours I could try to make the home page links a bit more compact:

I'm probably tagging the release before then. But I'm happy to rebuild and re-upload the docs with any improvements.

---

A table-like structure for this links makes a lot of sense.

**Comment by datapythonista** [comment]
will see if it's fast enough so I can have it now then

**Comment by datapythonista** [comment]
I opened #24928

**Comment by Dr-Irv** [comment]
@datapythonista I just saw this and the change in the docs at https://pandas.pydata.org/pandas-docs/stable/reference/index.html.  There is one issue for me is that it is now MUCH harder to find the API specifics of an individual method.  In the previous docs with 0.23.4, I could go to that page and use the browser search to find a specific method (since I knew all the methods were listed) and then just click on the docs for that method to see the details of the signature.  

So having an index of ALL of the functions is really needed.  Or maybe it is there and I can't find it.

If this should be a new issue, let me know.

**Comment by TomAugspurger** [comment]
Joris raised this point earlier I think. You might try using the sphinx
search bar.

**Comment by Dr-Irv** [comment]
@TomAugspurger Fair enough.  I still think having that one list was useful.  As I'm working on projects, I will see how it affects my workflow.

**Comment by datapythonista** [comment]
The sphinx search option returned as the first option the correct method for any search like "Series.fillna", "Index.asof", "read_csv or pandas.read_csv"...

If you prefer searching in a huge document, you can also use the pdf, that would be somehow similar to what we had until now. (but I'm sure using the search option will be more convenient).

**Comment by Dr-Irv** [comment]
@datapythonista The other thing I've done is when I'm offline (e.g., on a plane), I can pull up the docs on my local machine, and search them using browser search, and I don't have the sphinx search there.

**Comment by datapythonista** [comment]
Sphinx search is client-side javascript only, if I'm not wrong. I think it should work locally.

**Comment by Dr-Irv** [comment]
@datapythonista You're right. Thanks for the pointer.  Still would like to see the search box at the top right of the page  :-)

**Pull request merged**